components/Comment/modules/annotation/lib/repositionAnnotations.ts

```diff
@@ -0,0 +1,144 @@
+    _annotationsSortedByY[focalPointIndex].threadCoordinates.x =
+      -SELECTED_THREAD_X_OFFSET;
+
+    repositioned[focalPoint.threadId] = _annotationsSortedByY[focalPointIndex];
+  }
+
+  for (let i = focalPointIndex; i >= 0; i--) {
+    const annotation = _annotationsSortedByY[i];
+    const prevAnnotation = _annotationsSortedByY[i - 1];
+    const threadRef = threadRefs[i];
+    const prevThreadRef = threadRefs[i - 1];
+
+    if (prevThreadRef?.current && threadRef?.current) {
+      const prevRect = prevThreadRef.current.getBoundingClientRect();
+      const prevRectBottom =
+        prevAnnotation.threadCoordinates.y + prevRect.height;
+
+      if (
+        annotation.threadCoordinates.y <
+        prevRectBottom + MIN_SPACE_BETWEEN_THREAD
+      ) {
+        const prevThreadNewPosY =
+          annotation.threadCoordinates.y -
+          MIN_SPACE_BETWEEN_THREAD -
+          prevRect.height;
+        if (prevAnnotation.threadCoordinates.y !== prevThreadNewPosY) {
+          // console.log('Current annotation:', annotation.serialized.textContent)
+          // console.log('Prev annotation:', prevAnnotation.serialized.textContent)
+          console.log(
+            `Updating [Before] Position of thread ${i - 1} from ${
+              prevAnnotation.threadCoordinates.y
+            } to: ${prevThreadNewPosY}`
+          );
+
+          // In addition to keeping track of repositioned, we need to update immediately by modifying current list so that the next iteration
+          // has correct value.
+          _annotationsSortedByY[i - 1].threadCoordinates.y = prevThreadNewPosY;
+
+          repositioned[prevAnnotation.threadId] = {
+            ...prevAnnotation,
+            threadCoordinates: {
+              ...prevAnnotation.threadCoordinates,
+              y: prevThreadNewPosY,
+            },
+          };
+        }
+      }
+    }
+  }
+
+  for (let i = focalPointIndex; i < _annotationsSortedByY.length - 1; i++) {
+    const annotation = _annotationsSortedByY[i];
+    const nextAnnotation = _annotationsSortedByY[i + 1];
+    const threadRef = threadRefs[i];
+    const nextThreadRef = threadRefs[i + 1];
+
+    if (nextThreadRef?.current && threadRef?.current) {
+      const currentRect = threadRef.current.getBoundingClientRect();
+      const nextRectTop = nextAnnotation.threadCoordinates.y;
+
+      // console.log('--------------------------------------------------------------')
+      // console.log('i', i)
+      // console.log('threadRefs', threadRefs)
+      // console.log('focusPointAndAfter', _annotationsSortedByY)
+      // console.log('threadRef', threadRef)
+      // console.log('current thread:', annotation.serialized.textContent)
+      // console.log("annotation.threadCoordinates.y:", annotation.threadCoordinates.y)
+      // console.log("currentRect.height:", currentRect.height)
+      // console.log('nextRectTop:', nextRectTop)
+      // console.log('next thread:', nextAnnotation.serialized.textContent)
```

Comment:
rm

Reply:
I deleted a bunch of this but kept a few lines I need for debugging the feature until it is fully out